src/Serilog.Sinks.RollingFile/Sinks/RollingFile/TemplatedPathRoller.cs

```diff
@@ -26,20 +27,35 @@ namespace Serilog.Sinks.RollingFile
     //
     class TemplatedPathRoller
     {
-        const string OldStyleDateSpecifier = "{0}";
-        const string DateSpecifier = "{Date}";
-        const string DateFormat = "yyyyMMdd";
+        
         const string DefaultSeparator = "-";
 
+        const string SpecifierMatchGroup = "specifier";
+        const string SequenceNumberMatchGroup = "sequence";
+
         readonly string _pathTemplate;
         readonly Regex _filenameMatcher;
+        readonly Specifier _specifier = null;
 
         public TemplatedPathRoller(string pathTemplate)
         {
             if (pathTemplate == null) throw new ArgumentNullException(nameof(pathTemplate));
-            if (pathTemplate.Contains(OldStyleDateSpecifier))
-                throw new ArgumentException("The old-style date specifier " + OldStyleDateSpecifier +
-                    " is no longer supported, instead please use " + DateSpecifier);
+
+            if (pathTemplate.Contains(Specifier.OldStyleDateToken))
```

Comment:
The next 13 lines could be encapsulated in `Specifier.TryGetSpecifier(pathTemplate, out specifier)`. If it returns false, `_specifier` can be set to `Specifier.Date` and the path modified. `TryGetSpecifier()` could then also be used to avoid the chain of three `if` statements to decide whether it's included in the directory name.